test/metabase/lib/card_test.cljc

```diff
@@ -78,16 +78,3 @@
                {:lib/type :metadata/column
                 :name     "count"}]
               (lib/returned-columns query))))))
-
-(deftest ^:parallel visible-columns-use-result--metadata-test
-  (testing "visible-columns should use the Card's `:result-metadata` (regardless of what's actually in the Card)"
-    (let [venues-query (lib/query
-                        (lib/composed-metadata-provider
-                         (lib.tu/mock-metadata-provider
-                          {:cards [(assoc (:orders lib.tu/mock-cards) :dataset-query lib.tu/venues-query)]})
-                         meta/metadata-provider)
-                        (:orders lib.tu/mock-cards))]
-      (is (= ["ID" "SUBTOTAL" "TOTAL" "TAX" "DISCOUNT" "QUANTITY" "CREATED_AT" "PRODUCT_ID" "USER_ID"]
-             (mapv :name (get-in lib.tu/mock-cards [:orders :result-metadata]))))
-      (is (= ["ID" "SUBTOTAL" "TOTAL" "TAX" "DISCOUNT" "QUANTITY" "CREATED_AT" "PRODUCT_ID" "USER_ID"]
-             (mapv :name (lib/visible-columns venues-query)))))))
```

Comment:
@camsaul Removing this test is questionable, but required for this PR to work.

Reply:
instead of removing this, can't you call this with the `include-implicitly-joinable-source-card` setting disabled?

Reply:
That can be done, if we don't really mean what the test was saying.